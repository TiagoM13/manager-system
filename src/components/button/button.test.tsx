import React from 'react';
import '@testing-library/jest-dom';

import { render } from '@testing-library/react';

import { Button } from '.';

describe('<Button />', () => {
  test('should render correctly component', () => {
    const { getByText } = render(<Button />);

    expect(getByText('Salvar')).toBeTruthy();
  });

  test('should render correctly classNames', () => {
    const { getByTestId } = render(<Button id="teste" />);

    expect(getByTestId('btn-teste')).toHaveClass('bg-sky-600 hover:bg-sky-500');
  });

  test('should render correctly colors variables in the button', () => {
    const { asFragment, getByTestId, getByText } = render(
      <Button id="teste" variable="danger" label="Delete" />,
    );

    expect(getByTestId('btn-teste')).toHaveClass('bg-red-600 hover:bg-red-500');
    expect(getByText('Delete')).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });
});
