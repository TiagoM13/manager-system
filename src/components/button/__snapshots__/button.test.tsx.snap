// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly colors variables in the button 1`] = `
<DocumentFragment>
  <button
    class="bg-red-600 outline-red-500 hover:bg-red-500 disabled:bg-red-700 rounded-lg py-2.5 px-3 font-medium text-sm text-white flex gap-2 items-center justify-center"
    data-testid="btn-teste"
  >
    Delete
  </button>
</DocumentFragment>
`;
